src/gui.cc

```diff
@@ -149,6 +149,8 @@ error GUI::DisplayError(const error &err) {
         ss << "You are offline (" << err << ")";
         if (kBackendIsDownError == err) {
             DisplayOnlineState(kOnlineStateBackendDown);
+        } else if (kCannotConnectError == err) {
+            DisplayOnlineState(kOnlineStateRateLimit);
```

Comment:
From what I see, we're treating all `kCannotConnectError` error as a Rate Limit error, which is incorrect.